3.assess_generalizability/nbconverted/roc_qc_importance_eval.py

```diff
@@ -118,82 +117,148 @@
-print(f"AUC Model 1: {aucNoQC}")
-print(f"AUC Model 2: {aucQC}")
+# Calculate ROC AUC for the MGH cell line
+aucNoQC_MGH = roc_auc_score(y_MGH_noQC, y_probs_MGH_modelNoQC)
+aucQC_MGH = roc_auc_score(y_MGH_QC, y_probs_MGH_modelQC)
+
+print(f"AUC Model 1: {aucNoQC_MGH}")
+print(f"AUC Model 2: {aucQC_MGH}")
 
 
 # ## Perform ROC AUC bootstrapping method for both QC and no QC models and data
 
-# In[7]:
+# In[8]:
 
 
-# No QC model
-scores_model1 = bootstrap_roc_auc(y_binary_no_QC, y_probs_modelNoQC)
+# No QC model iNFixion cell line
+scores_model1_iNFixion = bootstrap_roc_auc(y_iNFixion_noQC, y_probs_iNFixion_modelNoQC)
 
-# QC model
-scores_model2 = bootstrap_roc_auc(y_binary_QC, y_probs_modelQC)
+# QC model iNFixion cell line
+scores_model2_iNFixion = bootstrap_roc_auc(y_iNFixion_QC, y_probs_iNFixion_modelQC)
 
 # Compare distributions
-t_stat, p_value = ttest_ind(scores_model1, scores_model2)
+t_stat, p_value = ttest_ind(scores_model1_iNFixion, scores_model2_iNFixion)
 print(f"T-statistic: {t_stat}, P-value: {p_value}")
+print(f"Mean ROC AUC for Model No-QC for iNFixion cell line: {np.mean(scores_model1_iNFixion)}")
+print(f"Mean ROC AUC for Model QC for iNFixion cell line: {np.mean(scores_model2_iNFixion)}")
 
 
-# In[8]:
+# In[9]:
 
 
-print(f"Mean ROC AUC for Model No-QC: {np.mean(scores_model1)}")
-print(f"Mean ROC AUC for Model QC: {np.mean(scores_model2)}")
+# No QC model for MGH cell line
+scores_model1_MGH = bootstrap_roc_auc(y_MGH_noQC, y_probs_MGH_modelNoQC)
 
+# QC model for MGH cell line
+scores_model2_MGH = bootstrap_roc_auc(y_MGH_QC, y_probs_MGH_modelQC)
 
-# In[9]:
+# Compare distributions
+t_stat, p_value = ttest_ind(scores_model1_MGH, scores_model2_MGH)
+print(f"T-statistic: {t_stat}, P-value: {p_value}")
+print(f"Mean ROC AUC for Model No-QC: {np.mean(scores_model1_MGH)}")
+print(f"Mean ROC AUC for Model QC: {np.mean(scores_model2_MGH)}")
+
+
+# ## Generate plot comparing QC and no QC model on the iNFixion cell lines
+
+# In[10]:
+
+
+# Define darker colors for the mean lines
+darker_teal = mcolors.to_rgba("teal", 0.8)
+darker_orchid = mcolors.to_rgba("orchid", 0.8)
+
+plt.hist(scores_model1_iNFixion, bins=50, alpha=0.5, label="Model No-QC", color="teal")
+plt.hist(scores_model2_iNFixion, bins=50, alpha=0.5, label="Model QC", color="orchid")
+
+# Add vertical lines for the means
+plt.axvline(
+    np.mean(scores_model1_iNFixion),
+    color=darker_teal,
+    linestyle="dashed",
+    linewidth=2,
+)
+plt.axvline(
+    np.mean(scores_model2_iNFixion),
+    color=darker_orchid,
+    linestyle="dashed",
+    linewidth=2,
+)
+
+plt.legend(loc="upper left", fontsize=10)
+plt.xlabel("ROC AUC Score", fontsize=12)
+plt.ylabel("Frequency", fontsize=12)
+plt.title(
+    "Bootstrap ROC AUC Distributions For\nQC versus No QC Data",
+    fontsize=14,
+    fontweight="bold",
+)
+plt.grid(True, linestyle="--", alpha=0.6)
+plt.tight_layout()
+
+# save figure
+plt.savefig(f"{figure_path}/iNFixion_bootstrap_ROC_AUC_QC_versus_no_QC.png", dpi=600)
```

Comment:
can you add some indication that infixion was the original and mgh was the derivative? I don't think I'll remember in 3 months which is which. (it would also be helpful to add that detail to the title)

Reply:
No worries! I have added this to the titles!